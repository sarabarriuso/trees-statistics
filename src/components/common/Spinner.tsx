import React from 'react';
import { ReactComponent as SpinnerSvg } from '../../assets/images/spinner.svg';

export interface ISpinnerProps {
  caption: string;
}

const Spinner: React.FC<ISpinnerProps> = ({ caption }) => {
  const DEFAULT_CLASSNAME = 'spinner';

  return (
    <div className={DEFAULT_CLASSNAME}>
      <div className={`${DEFAULT_CLASSNAME}__wrap`}>
        <SpinnerSvg className={`${DEFAULT_CLASSNAME}__svg`} />
        <p>{caption}</p>
      </div>
    </div>
  );
};

export default Spinner;
