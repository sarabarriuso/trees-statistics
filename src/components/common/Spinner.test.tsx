import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Spinner from './Spinner';

export default describe('Spinner Tests', () => {
  it('Renders successfully', () => {
    // Arrange
    const caption = 'Spinner caption';

    // Act
    const tree = shallow(<Spinner caption={caption} />);

    //Assert
    expect(toJson(tree)).toMatchSnapshot();
  });
});
