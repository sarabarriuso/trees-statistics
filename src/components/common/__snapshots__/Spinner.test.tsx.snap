// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Tests Renders successfully 1`] = `
<div
  className="spinner"
>
  <div
    className="spinner__wrap"
  >
    <ForwardRef(SvgSpinner)
      className="spinner__svg"
    />
    <p>
      Spinner caption
    </p>
  </div>
</div>
`;
